### PR TITLE
Adds node auto-repair e2e tests.

### PR DESCRIPTION
**Pull request by krzysztof-jastrzebski**
This PR adds node auto-repair e2e tests.

### CODE REVIEW

**Comment by shyamjvs** [comment]
/assign @mikedanese @roberthbailey 
Could one of you take a look if this looks reasonable?

**Comment by krzysztof-jastrzebski** [comment]
/test pull-kubernetes-unit
--

**Comment by roberthbailey** [comment]
Assigning to @krousey.

File: test/e2e/lifecycle/node_auto_repairs.go [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+		"--instances="+nodeName,
+		"--zone="+framework.TestContext.CloudConfig.Zone).CombinedOutput()
+	if err != nil {
+		glog.Errorf("Failed to recreate node: %s", string(output))
+		return fmt.Errorf("failed to recreate node: %v", err)
+	}
+	return nil
+}
+
+func isNodeReady(c clientset.Interface, nodeName string) (bool, error) {
```
Comment (mwielgus) [inline test/e2e/lifecycle/node_auto_repairs.go]: Are you sure we have any library function that does this already?

Comment (krzysztof-jastrzebski) [reply test/e2e/lifecycle/node_auto_repairs.go]: I didn't manage to find it.

File: test/e2e/lifecycle/node_auto_repairs.go [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+const (
+	defaultTimeout = 3 * time.Minute
+	repairTimeout  = 20 * time.Minute
+)
+
+var _ = SIGDescribe("Node Auto Repairs [Slow]", func() {
```
Comment (krousey) [inline test/e2e/lifecycle/node_auto_repairs.go]: This needs a disruptive tag

**Review by krousey — CHANGES_REQUESTED** [review]

**Review by krousey — APPROVED** [review]

File: test/e2e/lifecycle/node_auto_repairs.go (hunk 2 above)
Comment (krzysztof-jastrzebski) [reply test/e2e/lifecycle/node_auto_repairs.go]: Done

**Pull request merged**